### PR TITLE
Update dependencies based on upstream repo

### LINKED ISSUE #38: Incorrect dependencies for v2 series (at least)
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

I've looked at several instances of pmdarima >=2.0.0, and in almost all of them, there are incorrect dependencies, as compared with the [requirements.txt file in the pmdarima repo](https://github.com/alkaline-ml/pmdarima/blob/v2.0.0/requirements.txt).  In particular, the numpy version in this package is higher than in the repo's requirements.txt, and the statsmodels version is not high enough.

### Installed packages

```shell
N/A
```

### Environment info

```shell
N/A
```

### PR DESCRIPTION
**Pull request by DManowitz**
Update dependencies based on upstream repo

Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes #38

Trying to fix issue with conda package's `numpy` requirement much higher than upstream package's `numpy` requirement.  Also, removed `cython` as run dependency since conda CI infrastructure already builds .pyx files and distributes resulting files.  Finally, noticed incorrect `statsmodels` dependency, so fixed that.

### CODE REVIEW

**Comment by DManowitz** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,23 +21,25 @@ requirements:
     - numpy >=1.21.2                         # [build_platform != target_platform]
     - pybind11                               # [build_platform != target_platform]
     - {{ compiler('c') }}
+    - {{ stdlib("c") }}
```
Comment (tomasvanpottelbergh) [inline recipe/meta.yaml:24]: This seems to have broken the build. Do you think it can be removed?

**Review by tomasvanpottelbergh — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (DManowitz) [reply recipe/meta.yaml:24]: According to [this](https://github.com/conda-forge/conda-forge.github.io/issues/2102#issuecomment-2225802928), it looks like I just need to re-render, which I will do.

**Comment by DManowitz** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (tomasvanpottelbergh) [reply recipe/meta.yaml:24]: Great, thanks a lot for the effort! I'll merge the PR.

**Pull request merged**